src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -183,6 +190,33 @@ export default class DeployFromSettingsController {
   }
 
   /**
+   * @returns {!angular.$q.Promise}
+   * @export
+   */
+  updateAvailableProtocols() {
```

Comment:
Yes, I've even written in my first comment to you that I think that until data is retrieved from the server combobox should be disabled or marked with some progress bar. The switch also. Another solution is doing this async as you proposed.

I was considering validator at first, but this didn't seem like something that fits to be 'validator'